crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -1,21 +1,46 @@
+//! The `state` module provides the ability to use world-wide, interdependent, finite state machines.
+//! 
+//! Most of the utilities around state involve running systems during transitions between states, or
```

Comment:
This paragraph belongs after the 3 types of states: that information is much more important.